src/orion/algo/mofa/mofa.py

```diff
@@ -0,0 +1,386 @@
+        olh_samples, self.cur_n_levels = sampler.generate_olh_samples(
+            roi_space, self.n_levels, self.strength, self.index, self.rng
+        )
+        self.n_trials = len(olh_samples)
+        olh_trials = sampler.generate_trials(olh_samples, roi_space)
+        return olh_trials
+
+    def seed_rng(self, seed):
+        """Seed the state of the random number generator.
+
+        Parameters
+        ----------
+        seed: int
+            Integer seed for the random number generator.
+
+        """
+        self.rng = np.random.RandomState(seed)
+
+    @property
+    def state_dict(self):
+        """Return a state dict that can be used to reset the state of the algorithm."""
+        state_dict = super().state_dict
+        state_dict["rng_state"] = self.rng.get_state()
+        state_dict["completed_trials"] = copy.deepcopy(self.completed_trials)
+        state_dict["duplicates"] = copy.deepcopy(self.duplicates)
+        state_dict["converged"] = self.converged
+        state_dict["roi_space"] = self.roi_space
+        state_dict["frozen_param_values"] = copy.deepcopy(self.frozen_param_values)
+        if hasattr(self, "current_trials_params"):
+            state_dict["current_trials_params"] = copy.deepcopy(
+                self.current_trials_params
+            )
+        return state_dict
+
+    def set_state(self, state_dict):
+        """Reset the state of the algorithm based on the given state_dict
+
+        :param state_dict: Dictionary representing state of an algorithm
+        """
+        super().set_state(state_dict)
+        self.seed_rng(0)
+        self.rng.set_state(state_dict["rng_state"])
+        self.completed_trials = copy.deepcopy(state_dict["completed_trials"])
+        self.duplicates = copy.deepcopy(state_dict["duplicates"])
+        self.converged = state_dict["converged"]
+        self.roi_space = copy.deepcopy(state_dict["roi_space"])
+        self.frozen_param_values = copy.deepcopy(state_dict["frozen_param_values"])
+        if "current_trials_params" in state_dict:
+            self.current_trials_params = copy.deepcopy(
+                state_dict["current_trials_params"]
+            )
+        elif hasattr(self, "current_trials_params"):
+            self.current_trials_params = []
+
+    def suggest(self, num):
+        """Suggest a `num`ber of new sets of parameters.
+
+        Draws points from a prepared set of samples from an orthonal Latin hypercube.
+
+        Parameters
+        ----------
+        num: int, optional
+            Number of trials to suggest. The algorithm may return less than the number of trials
+            requested.
+
+        Returns
+        -------
+        list of trials or None
+            A list of trials representing values suggested by the algorithm. The algorithm may opt
+            out if it cannot make a good suggestion at the moment (it may be waiting for other
+            trials to complete), in which case it will return None.
```

Comment:
```suggestion
        list of trials
            A list of trials representing values suggested by the algorithm. The algorithm may opt
            out if it cannot make a good suggestion at the moment (it may be waiting for other
            trials to complete), in which case it will return an empty list.
```

Reply:
Addressed in https://github.com/Epistimio/orion/pull/864/commits/b3f58efb4dc69ffc97b85c11b959110c69933d4e